### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by Nadav-01**

### CODE REVIEW

File: backend/app.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            
+        return output
+    
+
+    def put(self, short_url):
+        args = translation_put_args.parse_args()
+        user_id  = args["user_id"] if args["user_id"] != None else "anonymous"
+        long_url = args["long_url"]
+        
+        dictify = {short_url: {"short_url": short_url,
```
Comment (Nadav-01) [inline backend/app.py:78]: remove layer

File: backend/app.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            except exc.IntegrityError:
+                dictify[short_url]['long_url'] = f"ERROR - {short_url} already exists"
+                dictify[short_url]['user_id']  = "ERROR"
+                return dictify, 409
+                    
+        return dictify, 201 # TODO - Verify it actually worked
+    
+    def delete(self, short_url):
+        with engine.connect() as con:
+            con.execute(f"DELETE FROM {URL_TABLE} WHERE short_url = \"{short_url}\";")
```
Comment (Nadav-01) [inline backend/app.py:97]: TODO: what if short_url doesnt exists

File: backend/tests/test1.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            print("Removed DB...", end='')
+        except:
+            print(colored("Failed to clear DB", "red"))
+            print("Try to see if there're any python processes running")
+            exit()
+    else:
+        print("No DB found...", end='')
+    print(colored("[OK]", "green"))
+
+def init_db():
```
Comment (Nadav-01) [inline backend/tests/test1.py:48]: rename to something like 'run put requests'

File: backend/tests/test1.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    for dat in data:
+        res = requests.put(test_utils.BASE + "translate/" + dat['short_url'], dat)
+        assert res.status_code == 201
+        
+    print(colored("[OK]", "green"))
+
+def run_app():
+    global app_process
+    app_process = subprocess.Popen(["python", "./app.py"], shell=True)
+    sleep(1)
```
Comment (Nadav-01) [inline backend/tests/test1.py:60]: optional: check it runs successfully before announcing 'OK'

**Comment by Assaf-Alon** [comment]
test

File: backend/app.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    # @marshal_with(translation_resource_fields) # Serialize it to json format 
+    def get(self, short_url):
+        output = None
+        
+        with engine.connect() as con:
+            output = con.execute(f"SELECT * FROM {URL_TABLE} WHERE short_url = \"{short_url}\";")
+            listify = [dict(row) for row in output]
+            
+            if (len(listify) == 0):
+                # output = jsonify({short_url: {"short_url": short_url, "long_url": f"ERROR - {short_url} not found", "user_id": "ERROR"}})
```
[the remark this answers is not in the document]
Comment (Assaf-Alon) [reply backend/app.py:64]: removed

File: backend/app.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+# REST API methods for the translate page
+# {BASE_DOMAIN}/translate/<string:short_url>"
+class UrlTranslations(Resource):
+    # @marshal_with(translation_resource_fields) # Serialize it to json format 
+    def get(self, short_url):
+        output = None
+        
+        with engine.connect() as con:
+            output = con.execute(f"SELECT * FROM {URL_TABLE} WHERE short_url = \"{short_url}\";")
```
[the remark this answers is not in the document]
Comment (Assaf-Alon) [reply backend/app.py:60]: Maybe later, add it as an optional task

File: backend/app.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+            else:
+                dictify = {short_url: listify[0]}
+                output = jsonify(dictify)
+            
+        return output
+    
+
+    def put(self, short_url):
+        args = translation_put_args.parse_args()
+        user_id  = args["user_id"] if args["user_id"] != None else "anonymous"
```
[the remark this answers is not in the document]
Comment (Assaf-Alon) [reply backend/app.py:75]: Added task for input validation

File: backend/app.py (hunk 1 above)
Comment (Assaf-Alon) [reply backend/app.py:78]: Might look into it later

**Pull request merged**